masonry/src/widget/sized_box.rs

```diff
@@ -26,7 +26,7 @@ use crate::{
 pub enum BackgroundBrush {
     Color(Color),
     Gradient(Gradient),
-    PainterFn(Box<dyn FnMut(&mut PaintCtx)>),
+    PainterFn(Box<dyn FnMut(&mut PaintCtx, &mut Scene) + Send + Sync>),
```

Comment:
I think you could spare yourself from having to write mutex code by making PainterFn cloneable and deriving Clone on BackgroundBrush.

```suggestion
    PainterFn(Arc<dyn Fn(&mut PaintCtx, &mut Scene) + Send + Sync>),
```

Reply:
Yeah, I've thought about something similar, not sure if `FnMut` is really necessary here. I'd like to avoid Mutex locking in views as long as possible (I imagine there are few cases though where it's *really* necessary)

Reply:
Busy waiting for a lock is not used here. I used `try_lock` and defaulted to existing background if the lock cannot be acquired immediately. But, I think the mutex can't be already locked in practice as we do not build/rebuild concurrently?